File: backend/ruby_foundations/exercises/small_problems_101/easy_7/swap_case.rb
def swapcase(string)
  result = string.chars.map do |letter|
    case letter
    when /[a-z]/ 
      letter.upcase
    when /[A-Z]/
      letter.downcase
    else
      letter
    end
  end
  result.join
end

puts swapcase('CamelCase') == 'cAMELcASE'
puts swapcase('Tonight on XYZ-TV') == 'tONIGHT ON xyz-tv'

File: frontend/javascript/exercises/string_and_text_processing/swap_case.js
function isLowerCase(char) {
  return char >= 'a' && char <= 'z'
}

function isUpperCase(char) {
  return char >= 'A' && char <= 'Z'
}

function swapcase(string) {
  return string.split('').map(function (char) {
    if (isLowerCase(char)) {
      return char.toUpperCase();
    } else if (isUpperCase(char)) {
      return char.toLowerCase();
    } else {
      return char;
    }
  }).join('');
}

console.log(swapcase('CamelCase') === 'cAMELcASE');         // 'cAMELcASE'
console.log(swapcase('Tonight on XYZ-TV') === 'tONIGHT ON xyz-tv');// 'tONIGHT ON xyz-tv'